### PR TITLE
Implement CO line emission

### PR DESCRIPTION
**Pull request by zonca**
@giuspugl I started to import the code, still working on it, I'll ask you to review it once I'm done.

Can you also please create test output at NSIDE 16, just one line, both with the pregenerated clouds and with no clouds?

### CODE REVIEW

File: so_pysm_models/co_lines.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+class COLines:
+    def __init__(
+        self,
+        target_nside,
+        has_polarization=True,
+        line="10",
+        include_high_galactic_latitude_clouds=False,
+        polarization_fraction=0.001,
+        theta_high_galactic_latitude_deg=0.,
```
Comment (zonca) [inline so_pysm_models/co_lines.py]: @giuspugl shouldn't the default for theta be higher?

Comment (giuspugl) [reply so_pysm_models/co_lines.py]: yes I agree! we can set the default to  20 deg .

**Comment by giuspugl** [comment]
> @giuspugl I started to import the code, still working on it, I'll ask you to review it once I'm done.
> 
> Can you also please create test output at NSIDE 16, just one line, both with the pregenerated clouds and with no clouds?

Do you need also to include polarized emission?

**Comment by zonca** [comment]
Yes

**Comment by giuspugl** [comment]
> Yes
ok they are in the shared folder, 
CO10_TQUmaps_*

**Comment by zonca** [comment]
is the output in `K_CMB`?

File: so_pysm_models/tests/test_co.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    co = COLines(target_nside=16,
+        has_polarization=True,
+        line="10",
+        include_high_galactic_latitude_clouds=include_high_galactic_latitude_clouds,
+        polarization_fraction=0.001,
+        theta_high_galactic_latitude_deg=20.,
+        random_seed=1234567,
+        verbose=False,
+        run_mcmole3d=False
+    )
```
Comment (zonca) [inline so_pysm_models/tests/test_co.py]: @giuspugl the test doesn't pass. I tried also with theta=0, can you please take a look and see if you find a bug in my code?

You could also checkout this repository and run the code yourself, thanks.

**Comment by zonca** [comment]
also @giuspugl can you please make a pull request to this branch (`co`) with more documentation for the model added to `docs/models.rst`?

**Comment by zonca** [comment]
I thought more about how to integrate this to PySM, I think there is no elegant way, I would rather keep it external, a bit like the `SONoiseSimulator` https://github.com/simonsobs/mapsims/blob/master/mapsims/noise.py#L17 class.
The only issue is we have to convolve with the beam.
So we first run PySM, then we simulate CO, then we add it:

    pysm_map += so_pysm_models.COLines(target_nside=nside, line="10").signal()

The issue is that if there is no bandpass integration, PySM calls a component for example with `co.signal(10, 100, 1000)`, if there is bandpass it calls: `co.signal(100, 105, 110, 115, 120)` (no need to be equally spaced). So we should add the CO line in the second case but not the first. But no idea how to distinguish between the 2 cases. @bthorne93 any workaround I haven't thought about?

@giuspugl please add to the docs what beam is built into these maps.

**Comment by giuspugl** [comment]
Yes it is!

**Comment by giuspugl** [comment]
> is the output in `K_CMB`?

Yes it is

File: so_pysm_models/tests/test_co.py (hunk 2 above)
Comment (giuspugl) [reply so_pysm_models/tests/test_co.py]: Ok found the error and fixed it, now tests pass i am gonna make a pull request to this branch with  more documentation and the details on the beams.

**Comment by zonca** [comment]
@giuspugl download on the fly works,
I think it would be useful to also add the frequencies of each line in GHz as properties of the class, can you please do that?

**Pull request merged**